e2e/vfolder-explorer-modal.spec.ts

```diff
@@ -0,0 +1,220 @@
+    await modal.verifyFolderName(rwFolderName);
+
+    // 4. Verify file explorer loads without errors
+    await modal.verifyFileExplorerLoaded();
+
+    // 5. Verify Read & Write permission is displayed in folder details
+    await modal.verifyPermission('Read & Write');
+
+    // 6. Verify upload functionality is available (Write permission)
+    const uploadButton = await modal.getUploadButton();
+    await expect(uploadButton).toBeEnabled();
+
+    // 7. Verify create folder functionality is available (Write permission)
+    const createButton = await modal.getCreateFolderButton();
+    await expect(createButton).toBeEnabled();
+
+    // 8. Close modal
+    await modal.close();
+  });
+
+  test('User can view files but cannot upload to read-only VFolder', async ({
+    page,
+  }) => {
+    // 1. Create a VFolder with Read Only permissions
+    await createVFolderAndVerify(page, roFolderName, 'general', 'user', 'ro');
+
+    // 2. Verify folder shows "Read only" in the table
+    await navigateTo(page, 'data');
+    const folderRow = page.getByRole('row', { name: new RegExp(roFolderName) });
+    await expect(folderRow.getByText('Read only').first()).toBeVisible();
+
+    // 3. Open the VFolder in FolderExplorerModal
+    const modal = await openFolderExplorer(page, roFolderName);
+
+    // 4. Verify modal opens successfully
+    await modal.verifyFolderName(roFolderName);
+
+    // 5. Verify "Read only" mount permission is displayed in folder details
+    await modal.verifyPermission('Read only');
+
+    // 6. Verify file explorer loads
+    await modal.verifyFileExplorerLoaded();
+
+    // 7. Close modal
+    await modal.close();
+  });
+
+  test('User sees error message when accessing non-existent VFolder', async ({
+    page,
+  }) => {
+    // 1. Navigate directly to a non-existent VFolder URL
+    await page.goto('http://127.0.0.1:9081/data?folder=non-existent-id-12345');
+
+    // Wait for modal to appear
+    const modal = new FolderExplorerModal(page);
+    await modal.waitForOpen();
+
+    // 2. Verify error message is displayed in modal
+    await modal.verifyErrorMessage('Folder not found or access denied');
+
+    // 3. Verify file explorer table is not within the modal body
+    // Check that BAIFileExplorer's table is not rendered
+    await modal.verifyFileExplorerNotLoaded();
+
+    // 4. Verify modal can still be closed
+    const closeButton = await modal.getCloseButton();
+    // For React development mode, the button may be covered by Red screen; use force click
+    await closeButton.click({ force: true });
+  });
+});
+
+test.describe.serial('FolderExplorerModal - User Modal Interaction', () => {
+  const testFolderName = 'e2e-test-folder-layout-' + new Date().getTime();
+
+  test.beforeAll(async ({ page }) => {
+    // Create the folder once before all tests in this serial group
+    await loginAsUser(page);
+    await createVFolderAndVerify(page, testFolderName);
+  });
```

Comment:
we can't use page fixture in `beforeAll`​.

```suggestion
  test.beforeAll(async ({ page }) => {
    // Create the folder once before all tests in this serial group
    const context = await browser.newContext();
    const page = await context.newPage();
    await loginAsUser(page);
    await createVFolderAndVerify(page, testFolderName);
  });
```

Reply:
I changed and submited. Please check again. Thank you.